driver/chain_syncer/chain_syncer_test.go

```diff
@@ -46,15 +72,71 @@ func (s *ChainSyncerTestSuite) TestSync() {
 	s.Nil(s.s.Sync(head))
 }
 
-// func (s *ChainSyncerTestSuite) TestSyncTriggerBeaconSync() {
-// 	s.s.p2pSyncVerifiedBlocks = true
-// 	s.s.state.setLatestVerifiedBlockHash(common.Hash{})
-// }
+func (s *ChainSyncerTestSuite) TestAheadOfProtocolVerifiedHead2() {
+	s.TakeSnapshot()
+	// propose a couple blocks
+	testutils.ProposeAndInsertEmptyBlocks(&s.ClientTestSuite, s.p, s.s.calldataSyncer)
+
+	// NOTE: need to prove the proposed blocks to be verified, writing helper function
+
+	// generate transactopts to interact with TaikoL1 contract with.
+	privKey, err := crypto.ToECDSA(common.Hex2Bytes(os.Getenv("L1_PROVER_PRIVATE_KEY")))
+	s.Nil(err)
+	opts, err := bind.NewKeyedTransactorWithChainID(privKey, s.RpcClient.L1ChainID)
+	s.Nil(err)
+
+	head, err := s.RpcClient.L1.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	l2Head, err := s.RpcClient.L2.HeaderByNumber(context.Background(), nil)
+	s.Nil(err)
+
+	log.Info("L1HeaderByNumber head: %v\n", head.Number)
```

Comment:
Lets make it more go-ethereum style and not use `%v` in `github.com/ethereum/go-ethereum/log`, like: `log.Info("L1 head", "number", head.Number)`